### PR TITLE
feat(models): adding backend logic to collect model metadata

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

This only adds the backend logic to collect a model metadatas. See https://github.com/containers/podman-desktop-extension-ai-lab/pull/1345 for details.

The `@huggingface/gguf` is used.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

Related to https://github.com/containers/podman-desktop-extension-ai-lab/issues/1344
Required for https://github.com/containers/podman-desktop-extension-ai-lab/pull/1280#issuecomment-2196310464

### How to test this PR?

- [x] unit tests has been added

### CODE REVIEW

File: packages/backend/src/managers/modelsManager.ts [hunk 1]
```diff
@@ -422,4 +424,36 @@ export class ModelsManager implements Disposable {
...
+      }
+      console.log('result', result);
+      return result.metadata;
+    } catch (err: unknown) {
+      data['error'] = err;
+      console.error(err);
+      throw err;
+    } finally {
+      data['duration'] = performance.now() - before;
+      this.telemetry.logUsage('get-metadata');
```
Comment (axel7083) [inline packages/backend/src/managers/modelsManager.ts:456]:
```suggestion
      this.telemetry.logUsage('get-metadata', data);
```

File: packages/backend/src/managers/modelsManager.ts [hunk 2]
```diff
@@ -422,4 +424,36 @@ export class ModelsManager implements Disposable {
...
+      if (this.isModelOnDisk(modelId)) {
+        const modelPath = path.normalize(getLocalModelFile(model));
+        console.debug(`[getModelMetadata] reading model ${modelPath}`);
+        result = await gguf(modelPath, { allowLocalFile: true });
+      } else if (model.url) {
+        result = await gguf(model.url);
+      } else {
+        throw new Error('cannot get model metadata');
+      }
+      console.log('result', result);
```
Comment (axel7083) [inline packages/backend/src/managers/modelsManager.ts:448]:
```suggestion
```

**Review by lstocchi — APPROVED** [review]
LGTM

**Review by jeffmaury — APPROVED** [review]
Shouldn't the metadata be part of the model model (😄 ) and be lazily loaded ?

**Pull request merged**